### PR TITLE
Some v6 suggestions

### PR DESCRIPTION
**Pull request by SimonSchick**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change

This is a small selection of non breaking changes that will are related to upgrading to node v6 which is going EOL end of this month.

The ONLY breaking change here should be that we are dropping node v6.

I will probably create another PR for more breaking changes in v6 @eseliger and I are looking into potentially dropping bluebird.

### CODE REVIEW

File: lib/dialects/abstract/query-generator.js [hunk 1]
```diff
@@ -489,9 +483,7 @@ class QueryGenerator {
       - rawTablename, the name of the table, without schema. Used to create the name of the index
    @private
   */
-  addIndexQuery(tableName, attributes, options, rawTablename) {
-    options = options || {};
-
+  addIndexQuery(tableName, attributes, options = {}, rawTablename) {
```
Comment (frlinw) [inline lib/dialects/abstract/query-generator.js]: Just a suggestion, it's better if optionals parameters are explicits: `rawTablename = null`

File: lib/model.js [hunk 2]
```diff
@@ -269,7 +269,7 @@ class Model {
       };
     }
 
-    const existingAttributes = _.clone(this.rawAttributes);
+    const existingAttributes = { ...this.rawAttributes };
```
Comment (frlinw) [inline lib/model.js]: Is a clone really required here ?

Comment (SimonSchick) [reply lib/model.js]: Idk, trying not to change logic.

**Comment by SimonSchick** [comment]
Revert function defaults for now as it has too many implicit breaking changes.

**Comment by SimonSchick** [comment]
@frlinw @eseliger Unfortunately I cannot test postgres locally atm since my dev env died and docker doesn't want to work on my current one, if one of you could figure out what exactly broke here that'd be great!

**Comment by SimonSchick** [comment]
@sushantdhiman this now builds, I would really appreciate some feedback on this so we can move forward for v6 🙂

File: docs/upgrade-to-v6.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Upgrade to v5
```
Comment (eseliger) [inline docs/upgrade-to-v6.md]: v6

Comment (SimonSchick) [reply docs/upgrade-to-v6.md]: aaaaaa

**Pull request merged**